src/components/kubectl/logs.ts

```
@@ -0,0 +1,161 @@
+    if (!container) {
+        return;
+    }
+
+    getLogsForContainer(kubectl, podSummary, container.name, displayMode);
+}
+
+/**
+ * Gets the logs for a container in a provided pod, in a provided namespace, in a provided container.
+ */
+function getLogsForContainer(
+    kubectl: Kubectl,
+    podSummary: PodSummary,
+    containerName: string | undefined,
+    displayMode: LogsDisplayMode
+) {
+    let cmd = `logs ${podSummary.name}`;
+
+    if (podSummary.namespace) {
+        cmd = `${cmd} --namespace=${podSummary.namespace}`;
+    }
+
+    if (containerName) {
+        cmd = `${cmd} --container=${containerName}`;
+    }
+
+    if (displayMode === LogsDisplayMode.Follow) {
+        cmd = `${cmd} -f`;
+        kubectl.invokeInNewTerminal(cmd, `${podSummary.name}-${containerName}`);
+        return;
+    }
+
+    kubectl.invokeInSharedTerminal(cmd);
+}
+
+/**
+ * Searches for a pod yaml spec from the open document
+ * or from the currently selected namespace.
+ */
+async function logsForPod(kubectl: Kubectl, displayMode: LogsDisplayMode): Promise<void> {
+    const editor = vscode.window.activeTextEditor;
+
+    if (editor) {
+        return await logsForPodFromOpenDocument(kubectl, editor, displayMode);
+    }
+
+    return await logsForPodFromCurrentNamespace(kubectl, displayMode);
+}
+
+/**
+ * Finds a Pod from the open editor.
+ */
+async function logsForPodFromOpenDocument(kubectl: Kubectl, editor: vscode.TextEditor, displayMode: LogsDisplayMode) {
+    const text = editor.document.getText();
+    try {
+        const obj: {} = yaml.safeLoad(text);
+        if (!isPod(obj)) {
+            return;
+        }
+
+        const podSummary = {
+            name: obj.metadata.name,
+            namespace: obj.metadata.namespace
+        };
+
+        getLogsForPod(kubectl, podSummary, displayMode);
+        return;
+    } catch (ex) {
+        host.showErrorMessage("Could not retrieve logs for pod for active document.");
```

This seems wrong.  If I have, say, an application code file open, and ask for logs, I'd like to be able to select the pod running the application rather than have to close the document or open the pod YAML first.  Again, the old behaviour was to swallow the YAML parse exception and fall through to the 'select a container' case.